src/account/operations/balance.rs

```diff
@@ -189,3 +236,28 @@ impl AccountHandle {
         })
     }
 }
+
+pub(crate) fn add_balances(balances: Vec<AccountBalance>) -> crate::Result<AccountBalance> {
+    let mut total_balance: AccountBalance = Default::default();
+
+    for balance in balances {
+        total_balance.base_coin.total += balance.base_coin.total;
+        total_balance.base_coin.available += balance.base_coin.available;
+        total_balance.required_storage_deposit += balance.required_storage_deposit;
+        total_balance.nfts.extend(balance.nfts.into_iter());
+        total_balance.aliases.extend(balance.aliases.into_iter());
+        total_balance.foundries.extend(balance.foundries.into_iter());
+        for native_token_balance in &balance.native_tokens {
+            if let Some(total_native_token_balance) = total_balance
```

Comment:
You never really add the native tokens to `total_balance` ?

Reply:
Fixed